### PR TITLE
[Fix-12350] fix source not found error

### PR DESCRIPTION
**Pull request by xleoken**
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

We run worker in k8s, will meet `source not found` error when submit tasks.

image

And, I found someone meet the same error.

> Unlucky, Version 3.1.0 does not include a fix for this bug.

_Originally posted by @black-06 in https://github.com/apache/dolphinscheduler/issues/10938#issuecomment-1269309737_
      
### Version

3.1.x

### Fix

We should use `bash` as the base command.

https://www.cnblogs.com/qianxiaoPro/p/16218564.html

### CODE REVIEW

**Comment by black-06** [comment]
Maybe it's better to modify
`private static final String SH = "sh";` 
to  `private static final String BASH = "bash";`
in `ShellCommandExecutor.java`

**Comment by zhuangchong** [comment]
This problem has been fixed in pr #12180

**Comment by xleoken** [comment]
> Maybe it's better to modify `private static final String SH = "sh";` to `private static final String BASH = "bash";` in `ShellCommandExecutor.java`

Make sense, updated.

It will be more compatible if we use `bash source`, but it's also ok if we modify the BASH directly.

**Comment by xleoken** [comment]
> This problem has been fixed in pr #12180

Hi @zhuangchong, 

I meet the same bug, but that patch has been inactive for two weeks. And it also wont to fix another bug(`sudo`), it's better to fix the bug quickly. BTW, It doesn't matter to close this issue.

**Review by SbloodyS — APPROVED** [review]
LGTM

**Comment by zhuangchong** [comment]
> > This problem has been fixed in pr #12180
> 
> Hi @zhuangchong,
> 
> I meet the same bug, but that patch has been inactive for two weeks. And it also wont to fix another bug(`sudo`), it's better to fix the bug quickly. BTW, It doesn't matter to close this issue.

I found the part in PR #12180 more complete than what you modified.

**Comment by xleoken** [comment]
> > > This problem has been fixed in pr #12180
> > 
> > 
> > Hi @zhuangchong,
> > I meet the same bug, but that patch has been inactive for two weeks. And it also wont to fix another bug(`sudo`), it's better to fix the bug quickly. BTW, It doesn't matter to close this issue.
> 
> I found the part in PR #12180 more complete than what you modified.

yeah, that patch has been inactive for a long time, and I don't understand the `sudo` bug.

**Comment by xleoken** [comment]
cc @simsicon, are you still working on the issue? https://github.com/apache/dolphinscheduler/pull/12180

**Comment by simsicon** [comment]
> cc @simsicon, are you still working on the issue? #12180

Yes, I am waiting for the approval and the merge to the PR.

**Comment by simsicon** [comment]
> > > > This problem has been fixed in pr #12180
> > > 
> > > 
> > > Hi @zhuangchong,
> > > I meet the same bug, but that patch has been inactive for two weeks. And it also wont to fix another bug(`sudo`), it's better to fix the bug quickly. BTW, It doesn't matter to close this issue.
> > 
> > 
> > I found the part in PR #12180 more complete than what you modified.
> 
> yeah, that patch has been inactive for a long time, and I don't understand the `sudo` bug.

Basically the default sudo will reset all envs which just initialized by dolphinscheduler, so we need to pass `-E` to reserve them.

**Comment by xleoken** [comment]
thanks @SbloodyS for review, please rereview the https://github.com/apache/dolphinscheduler/pull/12180

ok @simsicon, I close the issue.

**Pull request closed without merging**
